modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/io/Sinks.scala

```diff
@@ -30,19 +30,41 @@ import com.permutive.pubsub.producer.Model.{ProjectId, Topic}
 import com.permutive.pubsub.producer.encoder.MessageEncoder
 import com.permutive.pubsub.producer.grpc.{GooglePubsubProducer, PubsubProducerConfig}
 
-import com.snowplowanalytics.snowplow.enrich.fs2.{ByteSink, Enrich}
+import com.snowplowanalytics.snowplow.enrich.fs2.{AttributedByteSink, AttributedData, ByteSink}
 import com.snowplowanalytics.snowplow.enrich.fs2.config.io.{Authentication, Output}
 
 object Sinks {
 
   /**
    * Set the delay threshold to use for batching. After this amount of time has elapsed (counting
-   * from the first element added), the elements will be wrapped up in a batch and sent. This
-   * value should not be set too high, usually on the order of milliseconds. Otherwise, calls
-   * might appear to never complete.
+   * from the first element added), the elements will be wrapped up in a batch and sent.
+   *
+   * If the source MaxQueueSize is small, then the queue could block between batches if
+   * DelayThreshold is too large.  If the source MaxQueueSize is sufficiently large, then the
+   * DelayThreshold should not ever cause blocking.
    */
   val DelayThreshold: FiniteDuration = 200.milliseconds
 
+  /**
+   * A batch of messages will be emitted to PubSub when the batch reaches 1000 messages.
+   * We use 1000 because it is the maximum batch size allowed by PubSub.
+   * This overrides the permutive library default of `5`
+   */
+  val PubsubMaxBatchSize = 1000L
```

Comment:
Eventually we'll have to make all these properties configurable. Beam allows us to configure them and it saved a day couple of times already.